lib/identity/stateTransitions/assetLock/proof/instant/validateInstantAssetLockProofStructureFactory.js

```diff
@@ -0,0 +1,70 @@
+const { InstantLock } = require('@dashevo/dashcore-lib');
+
+const instantAssetLockProofSchema = require('../../../../../../schema/identity/stateTransition/assetLock/proof/instantAssetLockProof.json');
+
+const convertBuffersToArrays = require('../../../../../util/convertBuffersToArrays');
+const InvalidIdentityAssetLockProofError = require('../../../../../errors/InvalidIdentityAssetLockProofError');
+const IdentityAssetLockProofMismatchError = require('../../../../../errors/IdentityAssetLockProofMismatchError');
+const InvalidIdentityAssetLockProofSignatureError = require('../../../../../errors/InvalidIdentityAssetLockProofSignatureError');
+
+/**
+ * @param {JsonSchemaValidator} jsonSchemaValidator
+ * @param {StateRepository} stateRepository
+ * @returns {validateInstantAssetLockProofStructure}
+ */
+function validateInstantAssetLockProofStructureFactory(
+  jsonSchemaValidator,
+  stateRepository,
+) {
+  /**
+   * @typedef {validateInstantAssetLockProofStructure}
+   * @param {RawInstantAssetLockProof} rawAssetLockProof
+   * @param {Transaction} transaction
+   * @param {number} outputIndex
+   */
+  async function validateInstantAssetLockProofStructure(
+    rawAssetLockProof,
+    transaction,
+    outputIndex,
+  ) {
+    const result = jsonSchemaValidator.validate(
+      instantAssetLockProofSchema,
+      convertBuffersToArrays(rawAssetLockProof),
+    );
+
+    if (!result.isValid()) {
+      return result;
+    }
+
+    let instantLock;
+    try {
+      instantLock = InstantLock.fromBuffer(rawAssetLockProof.instantLock);
+    } catch (e) {
+      const error = new InvalidIdentityAssetLockProofError(e.message);
+
+      result.addError(error);
+
+      return result;
+    }
+
+    if (instantLock.txid !== transaction.id) {
+      result.addError(new IdentityAssetLockProofMismatchError());
+
+      return result;
+    }
+
+    // TODO check if output is present and the same
```

Comment:
@Cofresi we should validate locked outputs against asset lock transaction output. Any ideas how to do that?

Reply:
you don’t need to check the outputs, because verifying the isLock signature [here](https://github.com/dashevo/js-dpp/pull/276/files#diff-8013817c1dda7429580c28d56851617a4af6ecc950c2dd68ca3f780f7f39633bR60) will verify the outputs implicitly by calculating the requestId [here](https://github.com/dashevo/dashcore-lib/blob/v0.19-dev/lib/instantlock/instantlock.js#L227-L241) and then using that requestId to select the correct signatory quorum.
If the outputs in the isLock were wrong, then the signature verification would not work.

Reply:
Ok, so I verify signature only then. Thank you!